src/main/webapp/app/exercises/shared/result/result.component.ts

```diff
@@ -225,7 +226,8 @@ export class ResultComponent implements OnInit, OnChanges {
      */
     buildResultString() {
         if (this.submission && this.submission.submissionExerciseType === SubmissionExerciseType.PROGRAMMING && (this.submission as ProgrammingSubmission).buildFailed) {
-            return this.translate.instant('artemisApp.editor.buildFailed');
+            const isManualResult = this.result?.assessmentType === AssessmentType.SEMI_AUTOMATIC;
```

Comment:
```suggestion
            const isManualResult = this.result?.assessmentType !== AssessmentType.AUTOMATIC;
```

Reply:
this is better to keep it compatible with older exercises

Reply:
Good catch! 👍